src/components/KymaModules/CommunityModulesAddModule.js

```diff
@@ -0,0 +1,305 @@
+import ModulesCard from 'components/KymaModules/components/ModulesCard';
+
+import { useModulesReleaseQuery } from './kymaModulesQueries';
+import { CommunityModuleContext } from './providers/CommunityModuleProvider';
+import { useUploadResources } from 'resources/Namespaces/YamlUpload/useUploadResources';
+import { getModuleResourcesLinks, getAllResourcesYamls } from './support';
+
+import './KymaModulesAddModule.scss';
+import { ModuleTemplatesContext } from './providers/ModuleTemplatesProvider';
+
+export default function CommunityModulesAddModule(props) {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const notification = useNotification();
+
+  const post = usePost();
+  const { installedCommunityModules, communityModulesLoading } = useContext(
+    CommunityModuleContext,
+  );
+  const { communityModuleTemplates: moduleTemplates } = useContext(
+    ModuleTemplatesContext,
+  );
+  const [selectedModules, setSelectedModules] = useState([]);
+  const [resourcesToAply, setResourcesToAply] = useState([]);
+  const [layoutColumn, setLayoutColumn] = useRecoilState(columnLayoutState);
+
+  const uploadResources = useUploadResources(
+    resourcesToAply,
+    setResourcesToAply,
+    () => {},
+    'default',
+  );
+
+  const { data: moduleReleaseMetas } = useModulesReleaseQuery({});
+
+  const modulesAddData = moduleTemplates?.items.reduce((acc, module) => {
+    const name = module.metadata.labels['operator.kyma-project.io/module-name'];
+    const existingModule = acc.find(item => {
+      return item.metadata.name === name;
+    });
+    const isAlreadyInstalled = installedCommunityModules.find(
+      installedModule => installedModule.name === name,
+    );
+
+    const moduleMetaRelase = moduleReleaseMetas?.items.find(
+      item => item.spec.moduleName === name,
+    );
+
+    if (!existingModule && !isAlreadyInstalled) {
+      moduleMetaRelase?.spec.channels.forEach(channel => {
+        if (!acc.find(item => item.name === name)) {
+          acc.push({
+            name: name,
+            channels: [
+              {
+                channel: channel.channel,
+                version: channel.version,
+                isBeta: moduleMetaRelase.spec.beta ?? false,
+                isMetaRelease: true,
+                resources: module.spec.resources,
+              },
+            ],
+            docsUrl: module.spec.info.documentation,
+            icon: {
+              link: module.spec?.info?.icons[0]?.link,
+              name: module.spec?.info?.icons[0]?.name,
+            },
+          });
+        } else {
+          acc
+            .find(item => item.name === name)
+            .channels.push({
+              channel: channel.channel,
+              version: channel.version,
+              isBeta: moduleMetaRelase.spec.beta ?? false,
+              isMetaRelease: true,
+              resources: module.spec.resources,
+            });
+        }
+      });
+    }
```

Comment:
This could be moved to a separate function and called dynamically in `ModulesAdd` and `CommunityModulesAdd`